### PR TITLE
Turns art into a component

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

See title

## Why It's Good For The Game

It's more fun this way.

## Changelog
:cl:
code: Turns art into a component.
/:cl:

### CODE REVIEW

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/components/art.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	if(isobj(parent))
+		RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_obj_examine)
+	else
+		RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_other_examine)
+	if(isstructure(parent))
+		RegisterSignal(parent, COMSIG_ATOM_ATTACK_HAND, .proc/on_attack_hand)
+	if(isitem(parent))
+		RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/apply_moodlet)
+
+/datum/component/art/proc/apply_moodlet(mob/living/M, impress)
```
Comment (ninjanomnom) [inline code/datums/components/art.dm]: It's not necessarily living
```suggestion
/datum/component/art/proc/apply_moodlet(mob/M, impress)
```

File: code/datums/components/art.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		if(GREAT_ART to INFINITY)
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "artgreat", /datum/mood_event/artgreat)
+		if (GOOD_ART to GREAT_ART)
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "artgood", /datum/mood_event/artgood)
+		if (BAD_ART to GOOD_ART)
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "artok", /datum/mood_event/artok)
+		if (0 to BAD_ART)
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "artbad", /datum/mood_event/artbad)
+
+/datum/component/art/proc/on_other_examine(datum/source, mob/living/M)
```
Comment (ninjanomnom) [inline code/datums/components/art.dm]:
```suggestion
/datum/component/art/proc/on_other_examine(datum/source, mob/M)
```

File: code/datums/components/art.dm [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	apply_moodlet(M, impressiveness)
+
+/datum/component/art/proc/on_obj_examine(datum/source, mob/living/M)
```
Comment (ninjanomnom) [inline code/datums/components/art.dm]:
```suggestion
/datum/component/art/proc/on_obj_examine(datum/source, mob/M)
```

File: code/datums/components/art.dm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	var/obj/O = parent
+	apply_moodlet(M, impressiveness *(O.obj_integrity/O.max_integrity))
+
+/datum/component/art/proc/on_attack_hand(datum/source, mob/living/M)
```
Comment (ninjanomnom) [inline code/datums/components/art.dm]:
```suggestion
/datum/component/art/proc/on_attack_hand(datum/source, mob/M)
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by Tlaltecuhtli** [comment]
if you put artistic stuff inside a display case  it should inherit its art component, also stuff like some lavaland loot/clockcult shit from lavaland ruind/ cups / mech toys should have it

**Comment by AnturK** [comment]
Photos/bust had no attack_hand element so you can pick them up without delay. You probably shouldn't be changing this. Unless this works correctly ?

**Comment by nemvar** [comment]
You can still pick them up without delay. The attack hand is only for structures.
```
if(isstructure(parent))
		RegisterSignal(parent, COMSIG_ATOM_ATTACK_HAND, .proc/on_attack_hand)

**Pull request merged**